### PR TITLE
Ensure that UI SDK does not depend on the modular notification classes

### PR DESCRIPTION
**Pull request by LukasPaczos**
## Description

Closes https://github.com/mapbox/mapbox-navigation-android/issues/3662.

### Implementation

I moved some of the implementation pieces required by both the `libnavigation-ui` and `libtrip-notification` to shared `utils` and `base` modules so that we can remove the hard link between the UI SDK and default trip notification implementations.

## Testing

I will publish a snapshot from this branch to verify the fix in a standalone project.

### CODE REVIEW

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/summary/SummaryModelTest.kt [hunk 1]
```diff
@@ -18,6 +18,7 @@ import org.junit.runner.RunWith
 import org.robolectric.RobolectricTestRunner
 import org.robolectric.annotation.Config
 import java.util.Calendar
+import kotlin.jvm.Throws
```
Comment (JunDai) [inline libnavigation-ui/src/test/java/com/mapbox/navigation/ui/summary/SummaryModelTest.kt:21]: NIT: I don't checkout this branch to verify. But is it unused import?

**Review by JunDai — APPROVED** [review]
thanks @LukasPaczos for addressing this issue 🙇 

let's 🚢  it!

File: libnavigation-ui/src/test/java/com/mapbox/navigation/ui/summary/SummaryModelTest.kt (hunk 1 above)
Comment (Guardiola31337) [reply libnavigation-ui/src/test/java/com/mapbox/navigation/ui/summary/SummaryModelTest.kt:21]: It's used down here https://github.com/mapbox/mapbox-navigation-android/blob/17312de630ea31bea4518853e29c6d560755cd8b/libnavigation-ui/src/test/java/com/mapbox/navigation/ui/summary/SummaryModelTest.kt#L104

**Review by Guardiola31337 — COMMENTED** [review]
We should make sure that `libtrip-notification` translations are moved to a new `libnavigation-base` Transifex project and everything is setup 👌 from our side. Similar to what we did here https://github.com/mapbox/mapbox-navigation-android/pull/2456 and https://github.com/mapbox/mapbox-navigation-android/pull/2649

**Comment by LukasPaczos** [comment]
Pushed the refactored files to Transifex :+1:

**Pull request merged**